tools/lib/logreader.py

```diff
@@ -58,7 +66,7 @@ def __init__(self, fn, canonicalize=True, only_union_types=False, sort_by_time=F
       dat = bz2.decompress(dat)
     elif ext == ".zst" or dat.startswith(b'\x28\xB5\x2F\xFD'):
       # https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#zstandard-frames
-      dat = zstd.decompress(dat)
+      dat = decompress_stream(dat)
```

Comment:
why did this change?

Reply:
The `zstd.decompress` function cannot  decode zstd files written in streaming mode, raising the error: 

> zstd.ZstdError: could not determine content size in frame header. 

This happens because streaming mode does not include the original content size in the frame header. it's necessary to use zstd's streaming decompression, which processes data incrementally without requiring the content size upfront.

Reply:
Is the decompression speed roughly the same?

Reply:
zstd.decompress() is slightly faster because it can read the decompressed size from the frame header, allowing it to allocate the buffer in one go.

On the other hand, stream_reader().read() works by processing the data in chunks and then combining them at the end. While it's a bit slower since it doesn't know the decompressed size upfront, it’s still quite efficient. https://github.com/indygreg/python-zstandard/blob/9eb56949b1764a166845e065542690942a3203d3/c-ext/decompressionreader.c#L543

Overall, even though we lose a bit of speed, the streaming approach is still better because it reduces log sizes, saves space, and is more efficient with memory.